### ISSUE TITLE
Azuregos enrage spell

### ISSUE THREAD

**Issue by cala** [cmangos/issues#1184]
I wanted to see if I can contribute to bring the completion level of the Azuregos SD2 script from 90% to 100% and I saw [this](https://github.com/cmangos/mangos-classic/blob/master/src/scriptdev2/scripts/kalimdor/boss_azuregos.cpp#L38):
`SPELL_ENRAGE = 23537,`

 I don't remember an enrage for Azuregos but I may be wrong. Moreover, I'm surprised by the ID of the spell. All spells ID for this boss are in the range 21100 - 22100 from its release in patch 1.3, except two of them in the low 23100 range that were added in a patch 1.4 to avoid a corpse run exploit. So having an enrage spell with 23537 as an ID means that either:
- the ability was added way later in another patch but I can't find a note about this
- the spell ID was changed after patch 1.5
- the spell ID is wrong
- there is no enrage

Also, the [enrage is triggered once the boss goes below 26% HP](https://github.com/cmangos/mangos-classic/blob/master/src/scriptdev2/scripts/kalimdor/boss_azuregos.cpp#L142). That's a weird treshold. Usually, hit points triggers are more like 5, 10, 15, 20, 25, 33, 50...
```c++

        // EnrageTimer
        if (!m_bEnraged && m_creature->GetHealthPercent() < 26.0f)
        {
            if (DoCastSpellIfCan(m_creature, SPELL_ENRAGE) == CAST_OK)
                m_bEnraged = true;
        }
```

I did a bit of digging in the old scriptdev2 repository history and look at a bunch of videos but found nothing enlighting. Old strategy guides do not mention an enrage either but enrages were more soft in the early days (that was probably the only "soft" boss ability in early days) so it may have been overlooked.

So I'd like to poke @Grz3s @evil-at-wow @xfurry (or anybody with data here) to see if you have sniffs that could confirm this spell ID for the enrage and possibly when it is triggered (though I believe that is not easy to extract from sniffs).
Thank you for your help.

**Comment by MrMarvin** [comment]
Btw, whats the shrinking effect in this https://youtu.be/27Qb3-of-xI?t=635 video? Smoothe, Tranq shot, shiv maybe?

**Comment by Tobschinski** [comment]
http://www.wowhead.com/item=10716/gnomish-shrink-ray

**Comment by evil-at-wow** [comment]
I don't think I've ever done the Azuregos fight. So I'm sorry @cala, but no data from me this time. In particular since he's no longer around either...

**Comment by cala** [comment]
> I don't think I've ever done the Azuregos fight.

Pity, you missed a fun one. Thank you for your reply. :)

**Comment by Lillecarl** [comment]
@cala probably someone who thought he was using integers when he made < 26.f.

If it was integer it would begin on 25, so changing it to <= 25.f should prolly be right for that bit.

**Comment by Qingle** [comment]
@cala after some research, it appears that Azuregos indeed didn't have an enrage mechanic.

"Azuregos had a number of interesting abilities that made him a challenging encounter in the open world. To prevent “graveyard zerg” tactics, players killed by him received a debuff that froze them in ice for 15 minutes if they tried to attack him again. Without an enrage mechanic, the boss sometimes took 30 minutes to die in early classic, so you weren’t always completely out of the fight. He’d also teleport himself and nearby players by a short distance, dropping aggro completely for everyone within teleport range. He had a short-range mana burn aura that casters had to beware of and a spell reflection shield. He was immune to arcane and highly resistant to frost damage. It was a fun, hectic fight, if also painfully long."

[Source](http://www.blizzardwatch.com/2016/08/30/wow-archivist-world-bosses-mon/+&cd=4&hl=en&ct=clnk&gl=cz)

Albeit it is an article from this year. But from multiple sources there is not a single one that says he has an Enrage.

**Comment by cala** [comment]
Thank you for this research. 👍 

I also did my own and found only one comment and one video mentioning that Azuregos gets angry. Both were from hunter soloing Azuregos and it was in TBC and WOTLK.
Giving the almost 1 Million HP of Azuregos, an enrage starting at 25% would mean that it would be enraged for 250 000 HP, that is a big amount of HP at a time were MC was the only raid and mostly broken.

So unless some can bring evidence that Azuregos had an enrage *in Classic*, I will comment out that part of the code in Classic core while letting a comment explaining why it was commented out.

**Comment by BeerfishWoW** [comment]
I have tried to find sources indicating Azuregos had an enrage in Classic, but failed to do so. Furthermore, I can't remember ever fighting this boss in vanilla and having to deal with an enrage timer. I think you're on the right track, cala.

**Comment by Qingle** [comment]
Few Videos for Research:

TBC:
- https://www.youtube.com/watch?v=FxoS4BEiyh8
- https://www.youtube.com/watch?v=i_-_3tfmU-o

WoTLK:
- https://www.youtube.com/watch?v=RAKJP9jjBsc
- https://www.youtube.com/watch?v=256-6dpzBZA

None of these videos show an enrage.

**Comment by Lillecarl** [comment]
https://www.youtube.com/watch?v=27Qb3-of-xI <- Vanilla video, quality is shit... But it might help you guys.

**Comment by cala** [comment]
Thank you @Lillecarl I saw this one already along with one that had a (very blurry) combat log and no enrage were found. I'm pushing the change now.
Thanks also @Qingle for his research.

**Comment by cala** [comment]
Closed by https://github.com/cmangos/mangos-classic/commit/39c87707b089ff0c164800e906c6204d051e902a

**Comment by Lillecarl** [comment]
@cala https://github.com/blog/1439-closing-issues-across-repositories
this closes cmangos/issues#1184 ;)
